snapcraft/how-to/quickstart/how-snapcraft-builds-a-snap.md

```diff
@@ -0,0 +1,90 @@
+* It allows the developer to register and log into the Snap Store.
+* It can upload snaps to the Snap Store.
+* It can promote snaps to different channels.
+
+The list of available global options and commands can be checked with one of:
+
+```bash
+snapcraft help
+snapcraft help --all
+```
+
+<h2 id='heading--snapcraft'>snapcraft.yaml</h2>
+
+Snaps are created using a build recipe defined in a file called `snapcraft.yaml`.
+
+When the snapcraft tool is executed on the command line, it will look for the file in the current project work directory, either in the top-level folder or a `snap` subdirectory. If the file is found, snapcraft will then parse its contents and progress the build toward completion.
+
+`snapcraft.yaml` is a configuration file written in the YAML language, with stanzas defining the application structure and behavior. When snapcraft runs and parses this file, it will use the declared information to build the snap. For developers more familiar with the traditional Linux build systems, the process is somewhat similar to the use of a Makefile or an RPM spec file.
+
+You can create the `snapcraft.yaml` file manually, or you can run the `snapcraft init` command to create a template file in the `snap` subdirectory.
+
+```bash
+snapcraft init
+```
+
+A generated template file contains just enough data to build, split across three stanzas:
+
+* Metadata.
+* Confinement level.
+* Build definition.
+
+<h3 id='heading--definitions'>Main definitions inside snapcraft.yaml</h3>
+
+There is no one way for how a snap ought to be assembled. However, most `snapcraft.yaml` files have the same common elements, including a number of mandatory declarations. Below is a short list of these keys, which will be further explained in the Examples sections later in the tutorial.
+
+* **Metadata** - describes the snap functionality and provides identifiers by which the snap can be cataloged and searched in the Snap Store.
+* **Security confinement** - describes the level of security of the snap.
+* **Base** - describes which set of libraries the snap will use for its functionality. The base also defines the operating system version for the snap build instance in the virtual machine or container launched by Snapcraft. For instance, base: core18 means that Snapcraft will launch an Ubuntu 18.04 virtual machine or container, the set of tools and libraries used inside the snap will originate from the Ubuntu 18.04 repository archives, and the snap applications will “think” they are running on top of an Ubuntu 18.04 system, regardless of what the actual underlying Linux distribution is.
+* **Parts** - describes the components that will be used to assemble the snap.
+* **Apps** - describes the applications and their commands that can be run in an installed snap.
+
+It is important to note several additional details:
+
+* A snap may contain one or more parts.
+* A snap may contain one or more applications
```

Comment:
"None or more", especially for apps -- snaps like `mesa-core22` (and earlier/later iterations) don't include any apps, but merely *serve content* via a content interface.